template/README.md

```diff
@@ -11,27 +11,97 @@ If you found this in `fredrikhgrelland/vagrant-hashistack`, you may be intereste
 
 Documentation on [parent repository](https://github.com/fredrikhgrelland/vagrant-hashistack#usage).
 
-## Customizing and using the vagrant box
+## Goals of this template
+This template aims to supply a developer with the tools necessary to do three things:
+- Build a docker image
+- Build a terraform module
+- Test and develop these two within a hashistack ecosystem
+
+To achieve this we utilise a vagrant-box built from [fredrikhgrelland/vagrant-hashistack](https://github.com/fredrikhgrelland/vagrant-hashistack/).
+
+## Where do I start?
+First of all I would recommend you read the section ["Using the template and box"](using-the-template-and-box) for a detailed look at the box's features and configurations. Below is a rough step-by-step going from a fresh template to a functioning terraform module. You can find example-code of everything below in [test_example/](test_example/).
+
+Step-by-step:
+1. Create a [Dockerfile](https://docs.docker.com/engine/reference/builder/) and put it in `docker/`.
+1. Use the box's startup playbooks to build and make the image available within the box. [Startup scheme](default-startup-scheme). [Example building image](test_example/dev/ansible/01_build_docker_image.yml).
+1. Create a [Nomad job specification](https://www.nomadproject.io/docs/job-specification). This should be placed in [conf/nomad/](conf/nomad/README.md). [Example hcl-file](test_example/conf/nomad/countdash.hcl)
+1. Use terraform to deploy this job to Nomad. Refer to ["Terraform module requirements"](terraform-module-requirements) on how to do this.
+
+## File structure
+- [conf/nomad/](conf/nomad/), nomad job specifications
+- [dev/](dev/), files used to control testing and development
+  - [dev/ansible](dev/ansible), playbooks run at startup
+  - [dev/vagrant/conf/](dev/vagrant/conf), files altering the vagrant-box  
+- [docker/](docker/), files used to build docker-image
+- [example/](example/), terraform-files using the terraform-module, [ref](terraform-module-requirements)
+- [test_example/](test_example/), a copy of template, filled with example-code
+
+## Terraform module requirements
+[Hashiscorp documentation](https://www.terraform.io/docs/modules/index.html)
+
+ This template follows Hashicorp's standards for terraform modules. In short the root should contain a `main.tf`, `variables.tf` and `outputs.tf`. These three form a module. In addition to this there should be an `example/` folder that uses the module. To get a feel for how to use this effectively in our case you can look at the files [main.tf](test_example/main.tf), [variables.tf](test_example/variables.tf), and [outputs.tf](test_example/outputs.tf) from [test_example/](test_example/).
+
+## Using the template and box
 
```

Comment:
Draft, I would suggest moving `### Commands` section under here.
```suggestion

### Commands
There are several commands that help to run the vagrant-box:
- `make up` provisions a [vagrant-hashistack](https://github.com/fredrikhgrelland/vagrant-hashistack/) box on your machine. After the machine and hashistack are set up it will run the default startup scheme.

- `make clean` take down the provisioned box if there is any.

- `make update` downloads the newest version of the [vagrant-hashistack box](https://github.com/fredrikhgrelland/vagrant-hashistack/) from [vagrantcloud](https://vagrantcloud.com/fredrikhgrelland/hashistack).

- `make example` runs the the example in [test_example/](test_example/)

For full info, check `template/Makefile`

`!Note` - Makefile commands are not idempotent in the context of vagrant-box.  You could face the error of port collisions. Most of the cases it could happen because of the vagrant box has already been running.

Once vagrant-box is running, you can use other options to iterate of deployment in the development stage. 
```

Reply:
last sentence ref to #232

Reply:
Yeah, these are really good additions